### DISCUSSION TITLE
Get current user email in the flows step

### DISCUSSION THREAD

**Discussion by Eshaftri** [directus/directus#17401] [Archive]
Hi everyone,

I am encountering a problem where I want to retrieve the current user's `email` in one step and then add it to a collection field in the next step. Is this feasible using flows? I attempted to use the `$accountability.user.email` expression in the run script, but it was not successful.

Any suggestions would be greatly appreciated. Thank you!

**Comment by kulemantu** [comment]
One way that I have worked around this is by reading a user from a record by setting a field in that record that maps to the `directus_users` table to `$CURRENT_USER`, then retrieving that record and reading the user details via a relational field.

**An example:** 

A user is making an order by inserting a record into the `orders` data model that has `made_by` which maps to the user.

In a flow, you set `made_by` to `$CURRENT_USER` (it could really be any field that allows a user UUID, even temporarily, and set it back to `null` in a future flow). You can then retrieve the record via the **Read Data** flow and access `orders.made_by.first_name`, `orders.made_by.last_name` and `owned_by.email`, which should give you the user's details.

So in summary, for your use case, you may want to:
1. set a collection field e.g. `temp_user` to `$CURRENT_USER`, then 
2. Read the record and include a field called `collection.temp_user.email`
3. Update the record and set `temp_user` to `null` and set the `email` field on the collection to `{{previous_flow_name.temp_user.email}}`

❕ Very very hacky, but should work.

**Reply by kulemantu** [reply]
In the scenario that user logged in is the one who just updated the record, use `user_updated` relation field to get `user_updated.email`

**Reply by Eshaftri** [reply]
I'll definitely give it a go, thank you!

**Comment by BlackDahlia313** [comment]
> Hi everyone,
> 
> I am encountering a problem where I want to retrieve the current user's `email` in one step and then add it to a collection field in the next step. Is this feasible using flows? I attempted to use the `$accountability.user.email` expression in the run script, but it was not successful.
> 
> Any suggestions would be greatly appreciated. Thank you!

you have to read data and than call from that. :) so one extra step needed

**Comment by LevelUpGit** [comment]
> Hi everyone,
> 
> I am encountering a problem where I want to retrieve the current user's `email` in one step and then add it to a collection field in the next step. Is this feasible using flows? I attempted to use the `$accountability.user.email` expression in the run script, but it was not successful.
> 
> Any suggestions would be greatly appreciated. Thank you!

You can use $accountability.user to read user id, than get that user info from users collection, and you have access to email.
